src/olympia/addons/tests/test_commands.py

```diff
@@ -209,6 +209,24 @@ def test_process_addons_invalid_task():
         call_command('process_addons', task='foo')
 
 
+@pytest.mark.django_db
+def test_process_addons_limit_addons(monkeypatch):
+    def sign_all_addons_mock(ids):
+        assert ids == addon_ids
+
+    def sign_some_addons_mock(ids):
+        assert ids == addon_ids[:2]
+
+    addon_ids = [addon_factory().id for _ in range(5)]
+    assert Addon.objects.count() == 5
+
+    monkeypatch.setattr(SIGN_ADDONS, sign_all_addons_mock)
+    call_command('process_addons', task='sign_addons')
+
+    monkeypatch.setattr(SIGN_ADDONS, sign_some_addons_mock)
+    call_command('process_addons', task='sign_addons', limit=2)
+
```

Comment:
One thing I don't like about this test is that it doesn't prove that the function is called at all. Having something like a global counter incremented in each would solve that.

Reply:
https://github.com/mozilla/addons-server/pull/9699/commits/385d572c9fe0a4921c617162bfc377573d205d0a